api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2ClientConfig.java

```diff
@@ -49,7 +49,7 @@
  */
 @Value.Immutable
 @SuppressWarnings("immutables:subtype")
-abstract class OAuth2ClientConfig implements OAuth2AuthenticatorConfig {
+abstract class OAuth2ClientConfig implements OAuth2AuthenticatorConfig, AutoCloseable {
```

Comment:
Seems that making this config holding resources (adding `AutoCloseable`) adds a bunch of overhead (need to close the config in addition to the client).

I wonder whether it's worth to have a (lazily started) executor for each `HttpClient`, which then has the same lifecycle as the `HttpClient`. We can make that one have the same "thread-keep-alive" setting as the `OAuth2TokenRefreshExecutor`. WDYT?

Reply:
1. I agree that making `OAuth2ClientConfig` auto-closeable wasn't a great idea. Let's revert that and have `OAuth2Client` close relevant resources as before.
2. I like your idea of having a (lazily started) executor for each `HttpClient`. Can we tackle that separately though?

Reply:
> Can we tackle that separately though?
sure